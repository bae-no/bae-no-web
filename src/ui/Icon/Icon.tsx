import { useMemo } from "react";

import { Sprinkles, sprinkles } from "../sprinkles.css";
import { SvgIconKey, SVG_ICON_MAP } from "./iconMap";

interface IconProps extends Pick<Sprinkles, "color" | "size"> {
  name: SvgIconKey;
  "aria-label"?: string;
}

function Icon({ name, size, color = "black2", ...rest }: IconProps) {
  const IconComponent = useMemo(() => SVG_ICON_MAP[name], [name]);

  return <IconComponent className={sprinkles({ size, color })} {...rest} />;
}

export default Icon;
