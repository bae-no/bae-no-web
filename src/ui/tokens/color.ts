export const colors = {
  black1: "#121212",
  black10: "#F8F8F8",
  black11: "#E8E8E8",
  black2: "#313131",
  black3: "#666666",
  black4: "#888888",
  black5: "#A6A6A6",
  black6: "#CCCCCC",
  black7: "#DBDBDB",
  black8: "#EDEDED",
  black9: "#F2F2F2",
  danger1: "#FE5432",
  danger2: "#FE765B",
  danger3: "#FE9884",
  danger4: "#FFBBAD",
  danger5: "#FFDDD6",
  info1: "#1D86F4",
  info2: "#4A9EF6",
  info3: "#77B6F8",
  info4: "#A5CFFB",
  info5: "#D2E7FD",
  orange1: "#D9630B",
  orange2: "#FD8110",
  orange3: "#FD9A40",
  orange4: "#FEB36F",
  orange5: "#FECC9F",
  orange6: "#FFE6CF",
  orange7: "#EE790F",
  skyBlue1: "#33D1F3",
  success1: "#75D181",
  success2: "#91DA9A",
  success3: "#ACE3B3",
  success4: "#C8EDCD",
  success5: "#E3F6E6",
  warning1: "#F4DB33",
  warning2: "#F7E25C",
  warning3: "#F9E985",
  warning4: "#FBF1AD",
  warning5: "#FDF8D6",
  white: "#FFFFFF",
};
