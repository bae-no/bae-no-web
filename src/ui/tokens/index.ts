export * from "./color";
export * from "./layout";
export * from "./size";
export * from "./space";
