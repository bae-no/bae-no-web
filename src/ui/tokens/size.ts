export const size = {
  sm: "0.25rem",
  md: "0.5rem",
  lg: "0.75rem",
  xl: "1rem",
};
