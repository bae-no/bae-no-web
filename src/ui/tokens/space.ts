export const space = {
  "0": "0",
  "1": "0.1rem",
  "12": "1.2rem",
  "136": "13.6rem",
  "16": "1.6rem",
  "2": "0.2rem",
  "20": "2.0rem",
  "208": "20.8rem",
  "24": "2.4rem",
  "32": "3.2rem",
  "4": "0.4rem",
  "48": "4.8rem",
  "5.5": "0.55rem",
  "56": "5.6rem",
  "64": "6.4rem",
  "8": "0.8rem",
};
