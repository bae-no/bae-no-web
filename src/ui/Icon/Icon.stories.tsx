import { ComponentStory, ComponentMeta } from "@storybook/react";

import { colors } from "../tokens/color";
import { space } from "../tokens/space";

import Icon from "./Icon";

export default {
  argTypes: {
    color: { control: "select", options: Object.keys(colors) },
    size: { control: "select", options: Object.keys(space) },
  },
  component: Icon,
  title: "UI/Icon",
} as ComponentMeta<typeof Icon>;

const Template: ComponentStory<typeof Icon> = (args) => <Icon {...args} />;

export const Default = Template.bind({});
Default.args = {
  color: "black1",
  name: "arrow-left",
};
