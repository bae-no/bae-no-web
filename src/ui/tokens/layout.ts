import { space } from "./space";

export const layout = {
  auto: "auto",
  fit: "fit-content",
  full: "100%",
  half: "50%",
  max: "max-content",
  min: "min-content",
  ...space,
};
