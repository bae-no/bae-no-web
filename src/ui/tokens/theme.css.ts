import { createGlobalTheme } from "@vanilla-extract/css";

import { colors } from "./color";
import { layout } from "./layout";
import { space } from "./space";
import { transform } from "./transform";

export const theme = createGlobalTheme("body", {
  colors,
  layout,
  space,
  transform,
});
