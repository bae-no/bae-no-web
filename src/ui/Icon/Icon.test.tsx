import { render, screen } from "@testing-library/react";
import { SVGAttributes } from "react";
import { theme } from "../tokens";
import Icon from "./Icon";

jest.mock("next/dynamic", () => () => {
  function DynamicComponent(props: SVGAttributes<SVGElement>) {
    return <svg {...props} />;
  }
  DynamicComponent.displayName = "LoadableComponent";
  DynamicComponent.preload = jest.fn();
  return DynamicComponent;
});

describe("UI Icon Component", () => {
  it("match snapshot", () => {
    const { container } = render(<Icon name="close" data-testid="test" />);

    expect(container).toMatchSnapshot();
  });

  it("could have aria-label", () => {
    render(<Icon aria-label="test" name="close" />);
    expect(screen.getByLabelText("test")).toBeInTheDocument();
  });

  it("could have color", () => {
    const { container } = render(<Icon color="black1" name="close" />);
    expect(container).toHaveStyle({ color: theme.colors.black1 });
  });

  it("could have size", () => {
    const { container } = render(<Icon size="xxl" name="close" />);
    expect(container).toHaveStyle({
      width: theme.size.xxl,
      height: theme.size.xxl,
    });
  });
});
