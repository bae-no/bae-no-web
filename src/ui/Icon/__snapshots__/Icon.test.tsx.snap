// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI Icon Component match snapshot 1`] = `
<div>
  <svg
    class="sprinkles_color_black2_lightMode__1vna7gblf"
    data-testid="test"
  />
</div>
`;
